File: src/Runtime/Value/ValueBool.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {
	public class ValueBool : TypeValue<ValueBool> {
		private bool _value = false;
		public static readonly ValueBool TRUE = new ValueBool(true);
		public static readonly ValueBool FALSE = new ValueBool(false);

		public bool value { get => _value; }

		public ValueBool(bool value) {
			_value = value;
		}
		public override string ToString() {
			return _value ? "true" : "false";
		}
		public override Value Clone() {
			return _value ? TRUE : FALSE;
		}

	}
} // namespace peak.interpreter

File: src/runtime/value/value_bool.cpp
#include "value_bool.h"

using namespace peak;

const std::shared_ptr<ValueBool> ValueBool::TRUE = std::make_shared<ValueBool>(true);
const std::shared_ptr<ValueBool> ValueBool::FALSE = std::make_shared<ValueBool>(false);

ValueBool::ValueBool(bool value)
	: _value(value) {
}

bool ValueBool::GetValue() const {
	return _value;
}

std::string ValueBool::ToString() const {
	return _value ? "true" : "false";
}

std::shared_ptr<Value> ValueBool::Clone() const {
	return _value ? TRUE : FALSE;
}
